### PR TITLE
add fix for bin_env

### PR DESCRIPTION
**Pull request by cmcmarrow**
### What does this PR do?
fixs the arg bin_env so it can work with binary builds.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/55507

### Tests written?
yes

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by twangboy — APPROVED** [review]

**Comment by cmcmarrow** [comment]
re-run full ubuntu

**Comment by cmcmarrow** [comment]
re-run full windows
re-run full centos
re-run full amazon

**Comment by cmcmarrow** [comment]
re-run full windows
re-run full centos
re-run full amazon

**Comment by cmcmarrow** [comment]
re-run full centos

**Review by Akm0d — APPROVED** [review]

File: tests/unit/modules/test_pip.py [hunk 1]
```diff
@@ -14,12 +14,29 @@
...
 class PipTestCase(TestCase, LoaderModuleMockMixin):
     def setup_loader_modules(self):
         return {pip: {'__salt__': {'cmd.which_bin': lambda _: 'pip'}}}
 
+    def test__pip_bin_env(self):
+        ret = pip._pip_bin_env(None, 'C:/Users/ch44d/Documents/salt/tests/pip.exe')
+        if salt.utils.platform.is_windows():
+            self.assertEqual(ret, 'C:/Users/ch44d/Documents/salt/tests')
+        else:
+            self.assertEqual(ret, None)
```
Comment (s0undt3ch) [inline tests/unit/modules/test_pip.py]: `self.assertIsNone(ret)`

File: tests/unit/modules/test_pip.py [hunk 2]
```diff
@@ -14,12 +14,29 @@
...
+
+    def test__pip_bin_env_no_change(self):
+        cwd = 'C:/Users/ch44d/Desktop'
+        ret = pip._pip_bin_env(cwd, 'C:/Users/ch44d/Documents/salt/tests/pip.exe')
+        self.assertEqual(ret, cwd)
+
+    def test__pip_bin_env_no_bin_env(self):
+        ret = pip._pip_bin_env(None, None)
+        self.assertEqual(None, None)
```
Comment (s0undt3ch) [inline tests/unit/modules/test_pip.py]: `self.assertIsNone(ret)`?

**Pull request merged**